src/host/polyfills/simpleView.test.ts

```diff
@@ -186,27 +135,110 @@ describe("simpleView", () => {
             margin-left: -4px;
             flex: none;
         }`;
         const expectedResult =
-            ".tabbed-pane-right-toolbar {\\n            display: none !important;\\n        }";
+            ".tabbed-pane-right-toolbar {\\n            visibility: hidden !important;\\n        }";
         let fileContents = getTextFromFile("shell.js");
 
         // The file was not found, so test that at least the text is being replaced.
-        fileContents = fileContents ? fileContents : expectedCss;
+        fileContents = fileContents ? fileContents : comparableText;
         const apply = await import("./simpleView");
         const result = apply.applyInspectorCommonCssRightToolbarPatch(fileContents, true);
         expect(result).not.toEqual(null);
         expect(result).toEqual(expect.stringContaining(expectedResult));
     });
+
+    it("applyDrawerTabLocationPatch correctly changes text", async () => {
+        const apply = await import("./simpleView");
+
+        const comparableText = "this._showDrawer.bind(this, false), 'drawer-view', true, true";
+        let fileContents = getTextFromFile("ui/InspectorView.js");
+        // The file was not found, so test that at least the text is being replaced.
+        fileContents = fileContents ? fileContents : comparableText;
+        const result = apply.applyDrawerTabLocationPatch(fileContents);
+        expect(result).not.toEqual(null);
+        expect(result).toEqual(expect.stringContaining(
+            "this._showDrawer.bind(this, false), 'drawer-view', true, true, 'network.blocked-urls'"));
+    });
+
+    it("applyMainTabTabLocationPatch correctly changes text", async () => {
+        const apply = await import("./simpleView");
+
+        const comparableText = "InspectorFrontendHostInstance), 'panel', true, true, Root.Runtime.queryParam('panel')";
+        let fileContents = getTextFromFile("ui/InspectorView.js");
+        // The file was not found, so test that at least the text is being replaced.
+        fileContents = fileContents ? fileContents : comparableText;
+        const result = apply.applyMainTabTabLocationPatch(fileContents);
+        expect(result).not.toEqual(null);
+        expect(result).toEqual(expect.stringContaining(
+            "InspectorFrontendHostInstance), 'panel', true, true, 'network'"));
+    });
+
+    it("applyAppendTabPatch correctly changes text", async () => {
+        const apply = await import("./simpleView");
+        const comparableText = "appendTab(id, tabTitle, view, tabTooltip, userGesture, isCloseable, index) {";
+
+        let fileContents = getTextFromFile("ui/TabbedPane.js");
+        fileContents = fileContents ? fileContents : comparableText;
+        const result = apply.applyAppendTabPatch(fileContents);
+        expect(result).not.toEqual(null);
+        expect(result).toEqual(expect.stringContaining(
+            "appendTab(id, tabTitle, view, tabTooltip, userGesture, isCloseable, index) { if (id"));

```

Comment:
From my understanding these are how the tabs are generated:

Every tab/tool has a view associated with it, and when the ViewManager creates a tabbed section, it calls ViewManager.appendApplicable items to populate the tabs in the tabbedSection.  By filtering the tab ids at this level, the tabbed headers won't and cannot add unused tabs in their tab array property.

The previous patches were _showTabElement and _selectTab.
_showTabElement rendered the tabs within the tab header.  We can't render a tab that isn't added to the tab array, so we no longer need to filter here.

_selectTab previously would return early if a tab that wasn't on the exception list was clicked making it a noop.  This is what would cause tabs to be unselectable.  Since we are only appending tabs that we want, we do not need a check to make certain tabs unselectable.

Reply:
it still sounds to me that we need to completely understand the effect of this change. As an example:

inspector_main\InspectorMain.js, during SourcesPanelIndicator constructor:
self.UI.inspectorView.setPanelIcon('sources', icon);

in SetTabbedPane.js setTabIcon is defined as:
setTabIcon(id, icon) {
    const tab = this._tabsById.get(id);
    tab._setIcon(icon);
    this._updateTabElements();
  }

Will throw an exception as the id is not registered in the tab array . I tested this theory on an environment with the PR changes applied and got the same error:
![image](https://user-images.githubusercontent.com/43078681/80022570-1b37fd80-8491-11ea-993d-17dfef680dee.png)

Reply:
I just restored the applySelectTabPatch and applyShowTabElement functions and removed the appendTab patch.  In general, I don't love that there are so many hard coded tab ID references in the source code, but I guess we have to work around that in our patches.